src/java.base/share/classes/java/util/stream/ReferencePipeline.java

```diff
@@ -274,40 +277,41 @@ public final <R> Stream<R> flatMap(Function<? super P_OUT, ? extends Stream<? ex
                 StreamOpFlag.NOT_SORTED | StreamOpFlag.NOT_DISTINCT | StreamOpFlag.NOT_SIZED) {
             @Override
             Sink<P_OUT> opWrapSink(int flags, Sink<R> sink) {
-                return new Sink.ChainedReference<>(sink) {
-                    // true if cancellationRequested() has been called
-                    boolean cancellationRequestedCalled;
+                final boolean shorts = isShortCircuitingPipeline();
```

Comment:
Same as above (here having the two final aligned but having different meanings require some mental gymnastic)